nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -156,14 +189,15 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       Branch branch = client.getApi().commitMultipleOperations()
           .operation(Operation.Put.of(key, newTable, table))
           .commitMeta(NessieUtil.catalogOptions(builder, catalogOptions).build())
-          .branch(client.getRef().getAsBranch())
+          .branch(expectedHead)
```

Comment:
I believe you can now remove the ignore on https://github.com/apache/iceberg/blob/master/nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java#L156-L161 as the test would pass now